mso/resource_mso_schema_site_anp_epg_domain.go

```diff
@@ -152,31 +190,46 @@ func resourceMSOSchemaSiteAnpEpgDomainImport(d *schema.ResourceData, m interface
 	if err != nil {
 		return nil, fmt.Errorf("No Sites found")
 	}
-
-	stateSite := get_attribute[2]
+	templateInfo := regexp.MustCompile("-").Split(get_attribute[2], 2)
+	site_template := strings.Split(get_attribute[2], "-")
+	stateSite := site_template[0]
+	stateTemplate := templateInfo[1]
 	found := false
 	stateAnp := get_attribute[4]
 	stateEpg := get_attribute[6]
-	domain := get_attribute[8]
-	domainType := get_attribute[10]
 
 	var stateDomain string
 
-	if domainType == "vmmDomain" {
-		stateDomain = fmt.Sprintf("uni/vmmp-VMware/dom-%s", domain)
-
-	} else if domainType == "l3ExtDomain" {
-		stateDomain = fmt.Sprintf("uni/l3dom-%s", domain)
-
-	} else if domainType == "l2ExtDomain" {
-		stateDomain = fmt.Sprintf("uni/l2dom-%s", domain)
-
-	} else if domainType == "physicalDomain" {
-		stateDomain = fmt.Sprintf("uni/phys-%s", domain)
-
-	} else if domainType == "fibreChannelDomain" {
-		stateDomain = fmt.Sprintf("uni/fc-%s", domain)
-
+	vmmp_match, _ := regexp.MatchString(".*/uni/vmmp-.*", get_dn)
+	l3dom_match, _ := regexp.MatchString(".*/uni/l3dom-.*", get_dn)
+	l2dom_match, _ := regexp.MatchString(".*/uni/l2dom-.*", get_dn)
+	phys_match, _ := regexp.MatchString(".*/uni/phys-.*", get_dn)
+	fc_match, _ := regexp.MatchString(".*/uni/fc-.*", get_dn)
```

Comment:
You probably don't need all of those and can use the submatch of match_domain directly.

Reply:
ok